calendar/experiments/calendar/parent/ext-calendar-calendars.js

```diff
@@ -18,15 +18,22 @@ this.calendar_calendars = class extends ExtensionAPI {
       getResolvedCalendarById,
       isOwnCalendar,
       convertCalendar,
-    } = ChromeUtils.import("resource://ext-calendar-draft/api/ext-calendar-utils.jsm");
+    } = ChromeUtils.import(this.extension.rootURI.resolve("experiments/calendar/ext-calendar-utils.jsm"));
 
     return {
       calendar: {
         calendars: {
           query: async function({ type, url, name, color, readOnly, enabled }) {
             let calendars = calmgr.getCalendars();
 
-            let patterns = url ? parseMatchPatterns([url]) : null;
+            let patterns = null;
+            if (url) {
+              try {
+                patterns = new MatchPatternSet([url], { restrictSchemes: false });
```

Comment:
Why not use a single `MatchPattern` at this point?

Reply:
I recall that not working in the version of TB 78 that I tested with, but can't remember why it did not work. It is possible that the issue is by now fixed and was never affecting beta versions – in that case we could of course switch to `MatchPattern`. Did you test that by chance?

Reply:
I only tested that `new MatchPattern(url, { restrictSchemes: false })` appears to give equivalent results to `new MatchPatternSet([url], { restrictSchemes: false })` for a few sample patterns.